[Program Capture] higher order primitives support pytree input and output (#6081)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  [(#5967)](https://github.com/PennyLaneAI/pennylane/pull/5967)
  [(#5999)](https://github.com/PennyLaneAI/pennylane/pull/5999)

* During experimental program capture, the qnode call, `cond`, and `for_loop`

>   can all handle functions that accept and/ or return pytree structures.

Author: ```suggestion
  can all handle functions that accept and/ or return pytree structures.
  [(#6081)](https://github.com/PennyLaneAI/pennylane/pull/6081)
```

  [(#5999)](https://github.com/PennyLaneAI/pennylane/pull/5999)
  [(#6058)](https://github.com/PennyLaneAI/pennylane/pull/6058)

> * During experimental program capture, the qnode call, `cond`, and `for_loop`
>   can all handle functions that accept and/ or return pytree structures.

Reviewer: ```suggestion
* During experimental program capture, functions that accept and/or return `pytree` structures can now be handled in the `QNode` call, `cond`, `for_loop` and `while_loop`. 
```
